source/dataman/HDF5/AMHDF5File.cpp

```diff
@@ -169,3 +170,21 @@ bool AMHDF5File::flushHDF5File()
 	return true;
 }
 
+bool AMHDF5File::addGroup(const QString &groupName)
+{
+	AMHDF5Group* group = new AMHDF5Group(groupName);
+	if(group->createHDF5Group(id_))
```

Comment:
I'd redo these to be more like the following:

```c++
if (!group->createHDF5Group(id_)){
    group->deleteLater();
    AMErrorMon::alert();
    return false;
}

groups_ << group;

return true;
```

And the same would go with open.  You will also want to add a means of closing and getting those groups from the AMHDF5File class as well.